kolibri/core/kolibri_plugin.py

```diff
@@ -197,3 +198,26 @@ def language_font_file_tags(self):
     @property
     def plugin_data(self):
         return {"unsupportedUrl": reverse("kolibri:core:unsupported")}
+
+
+@register_hook
+class FrontendHeadFaviconHook(FrontEndBaseHeadHook):
+    """
+    Render a favicon link in the <head> tag of base.html, if a favicon is
+    provided by the theme.
+    """
+
+    @property
+    def head_html(self):
+        favicon_urls = [
+            logo["src"]
+            for logo in ThemeHook.get_theme().get("logos", [])
+            if logo.get("content_type", "") == "image/vnd.microsoft.icon"
+        ]
+
+        # Choose the first available .ico file. It's unlikely there's more than
+        # one specified in the theme.
+        if favicon_urls:
+            return '<link rel="shortcut icon" href="{}">'.format(favicon_urls[0])
+        else:
+            return ""
```

Comment:
I think it would also be helpful to always supply the default Kolibri icon here, in case it's not defined by the theme, that way we always have a favicon.